### PR TITLE
feat: update command-line options processing

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit updates the CLI processing funtion
within the valiator, along with associated tests.
Specifically, the "debug" and "print-validators"
options were removed.

Signed-off-by: Phil Adams <phil_adams@us.ibm.com>

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Dependencies have been updated as needed
- [x] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [ ] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [ ] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [ ] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [ ] Added tests for new rule (packages/ruleset/test/*.test.js)
- [ ] Added docs for new rule (docs/ibm-cloud-rules.md)

### CODE REVIEW

File: packages/validator/src/cli-validator/run-validator.js [hunk 1]
```diff
@@ -310,16 +309,10 @@ const processInput = async function(program) {
...
-        swagger,
-        configObject,
-        spectralResults,
-        debug
-      );
+      results = validator(logger, swagger, configObject, spectralResults);
     } catch (err) {
       logError(chalk, 'There was a problem with a validator.', getError(err));
-      if (debug) {
+      if (verbose) {
```
Comment (dpopp07) [inline packages/validator/src/cli-validator/run-validator.js:315]: The more I think about it, the more I like the idea of doing away with the verbose flag and using the log levels to distinguish these invocations but I think we can discuss that and do it in a separate PR because there are some other implications. Wanted to bring it up so I wouldn't forget but fine to leave it as is right now 👍

File: packages/validator/src/cli-validator/utils/get-copyright-string.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const getVersionString = require('./get-version-string');
+
+module.exports = function() {
+  return `IBM OpenAPI Validator (${getVersionString()}), @Copyright IBM Corporation 2017, 2023.\n`;
```
Comment (dpopp07) [inline packages/validator/src/cli-validator/utils/get-copyright-string.js:4]: I forgot to ask this in your previous PR but just out of curiosity - what was the motivation for including the copyright statement?

File: packages/validator/src/cli-validator/utils/print-results.js [hunk 3]
```diff
@@ -88,36 +81,38 @@ module.exports = function print(
...
-    (stats.errors.total ||
-      stats.warnings.total ||
-      stats.infos.total ||
-      stats.hints.total)
+    stats.errors.total ||
+    stats.warnings.total ||
+    stats.infos.total ||
+    stats.hints.total
   ) {
     logger.info(chalk.bgCyan('statistics\n'));
```
Comment (dpopp07) [inline packages/validator/src/cli-validator/utils/print-results.js:117]: Can we change this to be "summary" since that's the language we're using now?

```suggestion
    logger.info(chalk.bgCyan('summary\n'));
```

**Review by dpopp07 — APPROVED** [review]
Looks good! Left a suggestion and a question but approving to avoid re-review

File: packages/validator/src/cli-validator/utils/get-copyright-string.js (hunk 2 above)
Comment (padamstx) [reply packages/validator/src/cli-validator/utils/get-copyright-string.js:4]: Just from my experiences on various product teams, tools like that would typically display a copyright notice on the first line of output.  I think there might actually be some sort of guidelines around this topic that would tell us to log a copyright notice.
But I can easily take it out if it doesn't belong there.

Comment (dpopp07) [reply packages/validator/src/cli-validator/utils/get-copyright-string.js:4]: Oh I have no issue with it, I was just a little surprised. Sounds good to me

File: packages/validator/src/cli-validator/utils/print-results.js (hunk 3 above)
Comment (padamstx) [reply packages/validator/src/cli-validator/utils/print-results.js]: latest commit fixes this

**Review by dpopp07 — APPROVED** [review]
:shipit:
